### ISSUE TITLE
BUG: comparing TZ-aware timestamps to a scalar results in exception

### ISSUE THREAD

**Issue by chrisaycock** [pandas-dev/pandas#15966]
This works as expected:

```
pd.DataFrame({'test': [pd.Timestamp('2012-01-01 13:00:00'),
                       pd.Timestamp('2012-01-01 13:00:00')]}) == -1
```

However, this raises an exception:

```
pd.DataFrame({'test': [pd.Timestamp('2012-01-01 13:00:00+00:00'),
                       pd.Timestamp('2012-01-01 13:00:00+00:00')]}) == -1
```

**Comment by watercrossing** [comment]
This has also been fixed at some point, I can't reproduce on master.

**Comment by jreback** [comment]
can u put up a validation test to close?

**Comment by TomAugspurger** [comment]
Closed by #18170
